crates/node/src/networking/p2p/pea2pea.rs

```diff
@@ -279,14 +311,30 @@ impl Reading for P2P {
         tracing::debug!(parent: self.node().span(), "decrypted a message from {}", source);
 
         match bincode::deserialize(message.as_ref()) {
-            Ok(tx) => self.recv_tx(tx).await,
+            Ok(protocol::Message::V0(msg)) => match msg {
+                protocol::MessageV0::Transaction(tx) => self.recv_tx(tx).await,
+                protocol::MessageV0::DiagnosticsRequest(kind) => {
+                    self.process_diagnostics_request(kind).await?
+                }
+                // Nodes are expected to ignore the diagnostics response.
+                protocol::MessageV0::DiagnosticsResponse(_) => (),
+            },
```

Comment:
In the future, this should be extracted into separate function and later on possibly even into `protocol::Message::parse(bs)`.

Reply:
Yes, I think message protocol doesn't belong to the p2p domain. P2P is to exchange data between peers. It's more network and security related. Protocol data is to exchange data between node. It's more related to the node logic. 2 works, 2 separate domains.
For diagnostic, it's more about peer diag or node diag.